store/tikv/snapshot.go

```diff
@@ -67,7 +67,8 @@ type tikvSnapshot struct {
 	// cached use len(value)=0 to represent a key-value entry doesn't exist (a reliable truth from TiKV).
 	// In the BatchGet API, it use no key-value entry to represent non-exist.
 	// It's OK as long as there are no zero-byte values in the protocol.
-	cached map[string][]byte
+	cached  map[string][]byte
+	cacheMu sync.Mutex
```

Comment:
When will there be concurrency? 
Or just in case？

Reply:
I ran into concurrent map access panic, forgot where it is.